src/example/example.service.ts

```diff
@@ -0,0 +1,93 @@
+export class ExampleService {
+  public calculateAverage(numbers: number[]): number {
+    if (!numbers.length) return 0;
+    let sum = 0;
+    for (const num of numbers) {
+      sum += num;
+    }
+    return sum / numbers.length;
+  }
+
+  public getUserData(userId: string): string {
+    const query = `SELECT * FROM users WHERE id = '${userId}'`;
+    return this.executeQuery(query);
```

Comment:
PullProofAI

The `getUserData` method directly executes a SQL query without any parameterization or validation, which can lead to SQL injection vulnerabilities. It's crucial to use parameterized queries to prevent such security risks.

```diff
+ import { executeSafeQuery } from 'some-sql-library';

  public getUserData(userId: string): string {
    const query = `SELECT * FROM users WHERE id = ?`;
    return this.executeQuery(query, [userId]);
  }

+   private executeQuery(query: string, params: any[]): string {
+     console.log('Executing query:', query);
+     return executeSafeQuery(query, params);
+   }
```



<!-- This is an auto-generated comment by PullProofAI -->